src/Plugin/Action/BlockIP.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\rules\Plugin\Action\BlockIP.
+ */
+
+namespace Drupal\rules\Plugin\Action;
+
+use Drupal\ban\BanIpManagerInterface;
+use Drupal\Component\Utility\String;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\rules\Engine\RulesActionBase;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides the 'Block IP' action.
+ *
+ * @Action(
+ *   id = "rules_block_ip",
+ *   label = @Translation("Block IP"),
+ *   category = @Translation("System"),
+ *   context = {
+ *     "ip" = @ContextDefinition("string",
+ *       label = @Translation("IP Address")
+ *     )
+ *   }
+ * )
+ *
+ * @todo: Add access callback information from Drupal 7.
+ * @todo: We should maybe use a dedicated data type for the ip address, as we
+ * do in Drupal 7.
```

Comment:
Also, this misses a dependency on ban module. We do not have a way to specify that yet, but let's add a todo for it.